en/services/parameter.md

```diff
@@ -181,6 +181,10 @@ The drone may restart the sequence the `PARAM_VALUE` acknowledgment is not recei
 
 PX4 implements the protocol in a way that is compatible with this specification.
 
+PX4 additionally provides a mechanism that allows a GCS to *cache* parameters, which significantly reduces ready-to-use time for the GCS if parameters have not been changed since the previous parameter sync.
+The way that this mechanism works is that when the list of parameters is requested, PX4 first sends a `PARAM_VALUE` with the `param_index` of `INT16_MAX` (in code, referred to as `PARAM_HASH`) containing a *hash* of the parameter set. 
```

Comment:
Oh, is that how it works? And how is this hash calculated?

Reply:
I believe this is how it works, I was hoping you could confirm :-) 

Basically when you get the requestlist after you've already got the list at some point this sets [_send_all_index = PARAM_HASH]()https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_parameters.cpp#L95. The sender code (running all the time in a loop I "think" looks at this value and [sends the hash first](https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_parameters.cpp#L480)

The hash is calculated - from docs - as  "compute the CRC32 over all string param names and 4 byte values": https://github.com/PX4/Firmware/blob/master/src/lib/parameters/parameters.cpp#L1352

The rest of the parameters are then sent, but may be ignored by the GCS if the hash matches. In any case, it can immediately report params are up to date.

Reply:
I guess you're saying that this isn't sufficient for someone to implement without the hash? Is how I describe it above sufficient - perhaps with a link to the code?

Reply:
That would help yes.

Reply:
Thanks @julianoes . Done, can you please confirm this is resolved to your satisfaction and can be merged?

Reply:
When you get a chance.